### Table 2.15 classifies a sample of psychiatric patients by their diagnosis and by whether their treatment prescribed drugs.

##### a. Conduct a test of independence, and interpret the P-value.

In [1]:
#not oridinal
DD.table <- array(data=c(105,12,18,47,0,8,2,19,52,13), 
                 dim= c(5,2),
                 dimnames=list(Diagnosis=c("Schizophrenia","Affective.disorder","Neurosis","Personality.disorder","Special.symptoms"), 
                               Treatment=c("Drugs","No.Drugs")))

DD.table



XsqDD <- chisq.test(DD.table, correct=FALSE)
XsqDD$p.value

Treatment
Diagnosis              Drugs No.Drugs
  Schizophrenia        105    8      
  Affective.disorder    12    2      
  Neurosis              18   19      
  Personality.disorder  47   52      
  Special.symptoms       0   13

Warning message in chisq.test(DD.table, correct = FALSE):
“Chi-squared approximation may be incorrect”

[1] 2.254892e-17

The p value 2.254892e-17 is less than .05, therefore we have sufficient evidence to reject the null hypothesis that there is no association (ie independence) between their diagnosis and whether their treatment prescribed drugs

#### b. Obtain standardized residuals, and interpret.

In [2]:
XsqDD$stdres

Treatment
Diagnosis              Drugs     No.Drugs 
  Schizophrenia         7.874526 -7.874526
  Affective.disorder    1.602262 -1.602262
  Neurosis             -2.385315  2.385315
  Personality.disorder -4.841701  4.841701
  Special.symptoms     -5.139491  5.139491

Standardized residuals that have an absolute value that exceeds about 2 indicates that the null hypothesesis did not 'fit' that cell. We can see  for cells where the standardized residual is greater than 2, the number of people who has X diagnosis and Y treatment has been overpredicted. Cells where the standardized residual is less than -2 the number of people who has X diagnosis and Y treatement has been underpredicted. 

#### c. Partition chi-squared into three components to describe differences and similarities among the diagnoses, by comparing (i) the first two rows, (ii) the third and fourth rows, (iii) the last row to the first and second rows combined and the third and fourth rows combined.

In [3]:
#Calculate the G^2 statistc in the future. 
DD.table 
XsqDD

Two_rows<-DD.table[0:2,] #first two rows
Two_rows

Xsq_Tworows <- chisq.test(Two_rows, correct=FALSE)
Xsq_Tworows

Third_Fourth_Rows<-DD.table[c(3,4),] #third and fourth rows
Third_Fourth_Rows
Xsq_TF<- chisq.test(Third_Fourth_Rows, correct=FALSE)
Xsq_TF


Schizophrenia.Affective.disorder.D <-(105+12)
Schizophrenia.Affective.disorder.ND<-(8+2)

Neurosis.Personality.disorder.D<-(18+47)
Neurosis.Personality.disorder.ND<-(19+52)

#last row to combined first and second rows and third and fourth rows combined 

Combined.table <- array(data=c(Schizophrenia.Affective.disorder.D, Neurosis.Personality.disorder.D,0,Schizophrenia.Affective.disorder.ND,Neurosis.Personality.disorder.ND,13), 
                 dim= c(3,2),
                 dimnames=list(Diagnosis=c('Schizophrenia + Affective.disorder',
                              'Neurosis + Personality.disorder','Special.symptoms'), 
                               Treatment=c("Drugs","No.Drugs")))
Combined.table
Xsq_Combined<- chisq.test(Combined.table, correct=FALSE)
Xsq_Combined

Treatment
Diagnosis              Drugs No.Drugs
  Schizophrenia        105    8      
  Affective.disorder    12    2      
  Neurosis              18   19      
  Personality.disorder  47   52      
  Special.symptoms       0   13


	Pearson's Chi-squared test

data:  DD.table
X-squared = 84.188, df = 4, p-value < 2.2e-16


Treatment
Diagnosis            Drugs No.Drugs
  Schizophrenia      105   8       
  Affective.disorder  12   2

Warning message in chisq.test(Two_rows, correct = FALSE):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  Two_rows
X-squared = 0.8917, df = 1, p-value = 0.345


Treatment
Diagnosis              Drugs No.Drugs
  Neurosis             18    19      
  Personality.disorder 47    52


	Pearson's Chi-squared test

data:  Third_Fourth_Rows
X-squared = 0.014875, df = 1, p-value = 0.9029


Treatment
Diagnosis                            Drugs No.Drugs
  Schizophrenia + Affective.disorder 117   10      
  Neurosis + Personality.disorder     65   71      
  Special.symptoms                     0   13

Warning message in chisq.test(Combined.table, correct = FALSE):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  Combined.table
X-squared = 83.884, df = 2, p-value < 2.2e-16


The p value 0.345 for component 1 (first two rows) is greater than .05, which gives us evidence to not reject the null hypothesis that there is no association between Diagnosis and Treatement. 

The p value 0.9029 for component 2  (third and fourth rows) like component 1 is greater than .05, which gives us evidence to not reject the null hypothesis that there is no association between Diagnosis and Treatement. This p value is greater than the one for component 1 which indicates greater evidence to accept the null hypothesis. 

The p value < 2.2e-16 for component 3  (third and fourth rows) is less than .05, which gives us evidence to reject the null hypothesis that there is no association between Diagnosis and Treatement. 